src/Microsoft.DotNet.Build.Tasks/InstallPackageFromFile.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Build.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using Task = Microsoft.Build.Utilities.Task;
+using ThreadingTask = System.Threading.Tasks.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class InstallPackageFromFile : Task, ICancelableTask
+    {
+        [Required]
+        public string PackageFile { get; set; }
+
+        [Required]
+        public string PackagesFolder { get; set; }
+
+
+        private static readonly CancellationTokenSource TokenSource = new CancellationTokenSource();
```

Comment:
Following the pattern from https://github.com/dotnet/buildtools/blob/78260239e4db68ad5c2df774a27d5f9a54fd92e5/src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs#L26. Any potential concerns?